keps/sig-network/20191104-iptables-no-cluster-cidr.md

```diff
@@ -0,0 +1,372 @@
+	"-m", "conntrack",
+	"--ctstate", "RELATED,ESTABLISHED",
+	"-j", "ACCEPT",
+)
+writeLine(proxier.filterRules,
+	"-A", string(KubeForwardChain),
+	"-m", "comment", "--comment", `"kubernetes forwarding conntrack pod destination rule"`,
+	"-d", proxier.clusterCIDR,
+	"-m", "conntrack",
+	"--ctstate", "RELATED,ESTABLISHED",
+	"-j", "ACCEPT",
+)
+```
+Again, applying similar logic to the last rule for iptables, the proposal here is to simplify this drop reference to
+the proxy.clusterCIDR and just match on the connection state.
+
+### Risks and Mitigations
+
+The biggest risk we have is that we are expanding the scope of the last rule to potentially include non-kubernetes
+traffic. This is considered mostly safe as it does not break any of the intended drop behavior. Plus once the initial
+connection has been accepted, assuming nodes are used for kubernetes workloads, it's highly unlikely that we would
+need to not accept it later.
+
+## Design Details
+
+The idea of ‘determine cluster originated traffic’ would be captured in a new go interface type within kube-proxy,
+with different implementations of the interface. The kube-proxy implementation itself would just call method on
+interface to get the match criteria to write in the rule.
+
+The new behavior can be opted-in using two flags. The first to determine the mode to use for detection, and the
+other (optionally) being the value to use in that mode. This separation of mode and value has the nice property
+that if we default the mode to "cluster-cidr", then the current `--cluster-cidr` flag can be used as is to get
+the current behaviour. So upgrades with no changes retain current behavior.
+
+```
+--detect-local={cluster-cidr | node-cidr | pod-interface-prefix  | bridge}
+
+  the mode to use for detection local traffic. The default is cluster-cidr (current behavior)
+
+--cluster-cidr="cidr[,cidr,..]"
+
+  the current --cluster-cidr flag. It will be enhanced to read a comma separated list of CIDRs so that more
+  than one can be specified if necessary. kube-proxy considers traffic as local if source is one
+  of the CIDR values. This is only used if `--detect-local=cluster-cidr` .
+
+--node-cidr[="cidr[,cidr,..]"]
+
+  (optional) list of node CIDRs as a comma separated list. kube-proxy considers traffic as local if source is one
+  of the CIDR values. If value is not specified, or flag is omitted,  defaults to node.podCIDR property on the node.
+  This is only used if `--detect-local=node-cidr` .
+
+--pod-interface-prefix="prefix[,prefix,..]"
+
+  kube-proxy considers traffic as local if originating from an interface which matches one of given
+  prefixes. string argument is a comma separated list of interface prefix names, without the ending '+'.
+  This is only used if `--detect-local=pod-interface-prefix` or `--detect-local=bridge`. In the case of
+  latter, the prefix is used as option to `--physdev-in name` match instead of just `--physdev-in` in
```

Comment:
Do we really want a prefix?    I think this would be cleaner as a flag of its own.

`--pod-bridge-name`

Then, like above, we can document it as a trailing '+' meaning prefix.